src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java

```diff
@@ -0,0 +1,136 @@
+package org.spongepowered.common.data.processor.data;
+
+import com.google.common.base.Optional;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.pathfinding.PathPoint;
+import org.spongepowered.api.data.*;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.ImmutableTargetedLocationData;
+import org.spongepowered.api.data.manipulator.mutable.TargetedLocationData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.service.persistence.InvalidDataException;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+import org.spongepowered.common.Sponge;
+import org.spongepowered.common.data.DataProcessor;
+import org.spongepowered.common.data.manipulator.immutable.ImmutableSpongeTargetedLocationData;
+import org.spongepowered.common.data.manipulator.mutable.SpongeTargetedLocationData;
+import org.spongepowered.common.data.util.DataUtil;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class TargetedLocationDataProcessor implements DataProcessor<TargetedLocationData, ImmutableTargetedLocationData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof Entity;
+    }
+
+    @Override
+    public Optional<TargetedLocationData> from(DataHolder dataHolder) {
+        if(dataHolder instanceof EntityLiving) {
```

Comment:
@JBYoshi `TargetedLocationData` does have some javadocs as to the "expected" use cases:
```
An {@link DataManipulator} handling the supposed targetted
{@link Location}. Usually for the case of {@link EnderPearl}s, the
targeted {@link Location} is where the {@link EnderPearl} will
move towards until it's expiration time. In the case of
{@link ItemStack}s of type {@link ItemTypes#COMPASS}, the targeted
{@link Location} is where the compass will point towards.
```